src/App.tsx

```diff
@@ -1,24 +1,151 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Header } from './Components/Header';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { Notification } from './Components/Notiifcation';
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
+import { FilterType } from './types/Filter';
+import { ErrorMessage } from './types/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 13;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(FilterType.ALL);
+  const [errorMessege, setErrorMessege] = useState(ErrorMessage.NONE);
+  const [selectedId, setSelectedId] = useState(-1);
+  const [isLoadingCompleted, setIsLoadingCompleted] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    client.get<Todo[]>(`/todos?userId=${USER_ID}`)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessege(ErrorMessage.CANNOT_LOAD_TODOS);
+      });
+  }, []);
+
+  const addTodo = (title: string) => {
+    setErrorMessege(ErrorMessage.NONE);
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    });
+
+    client.post<Todo>('/todos', {
+      userId: USER_ID,
+      title,
+      completed: false,
+    })
+      .then(newTodo => {
+        setTodos((currentTodos) => [...currentTodos, newTodo]);
+      })
+      .catch((error) => {
+        setErrorMessege(ErrorMessage.UNABLE_TO_ADD_A_TODO);
+        throw error;
+      })
+      .finally(() => {
+        setTempTodo(null);
+      });
+  };
+
+  const deleteTodo = (todoId: number) => {
+    setErrorMessege(ErrorMessage.NONE);
+    setSelectedId(todoId);
+
+    client.delete(`/todos/${todoId}`)
+      .then(() => setTodos(todos.filter((todo) => todo.id !== todoId)))
+      .catch(() => {
+        setErrorMessege(ErrorMessage.UNABLE_TO_DELETE_A_TODO);
+      })
+      .finally(() => {
+        setSelectedId(-1);
+      });
+  };
+
+  const deleteAllCompleted = () => {
+    setIsLoadingCompleted(true);
+
+    const completedTodos = todos.filter((todo) => todo.completed);
+
+    const deletePromises = completedTodos.map((todo) => client.delete(`/todos/${todo.id}`));
+
+    Promise.all(deletePromises)
+      .then(() => {
+        setTodos(currentTodo => currentTodo.filter((todo) => !todo.completed));
+      })
+      .catch((error) => {
+        setErrorMessege(ErrorMessage.UNABLE_TO_DELETE_A_TODO);
+        throw error;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)

FYI